### PR TITLE
Add User Agent and HTTP_ACCEPT to recognisable session table

### PR DESCRIPTION
**Pull request by JonathanAndrews**
- The User Agent and HTTP_ACCEPT headers are now stored when users sign in.
- Both are also compared for recognition on sign in.
- Finally, the `security_level` can be set to `:strict`, `:normal` or `:relaxed`.

### CODE REVIEW

File: app/controllers/devise_recognisable/sessions_controller.rb [hunk 1]
```diff
@@ -2,28 +2,18 @@
...
-    #
-    # NOTE: Geocoder's location method might not be the safest?
-    # See https://github.com/alexreisner/geocoder#geocoding-http-requests
-    last_sign_in = Geocoder.search(previous_session.sign_in_ip).first
-    current_sign_in = Geocoder.search(request.location.ip).first
-    # NOTE: looks like sometimes the current_sign_in isn't a real thing?
-    distance = Geocoder::Calculations.distance_between(last_sign_in&.coordinates, current_sign_in&.coordinates)
-
-    if previous_session.sign_in_ip != request.location.ip or distance > Devise.max_ip_distance
+    unless Recogniser.recognise?(request, previous_session)
```
Comment (jalada) [inline app/controllers/devise_recognisable/sessions_controller.rb:16]: Ooooh I'm excited for this API.

**Comment by jalada** [comment]
Yeah I like this! 👍🏻

**Comment by JonathanAndrews** [comment]
@jalada Is this good to go into master as well?

**Comment by jalada** [comment]
Yup!

**Pull request merged**